pkg/admission/controller.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"fmt"
+	"log"
+	"reflect"
+	"sync"
+
+	admissionregistrationv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/kubernetes-sigs/controller-runtime/pkg/admission/certinput"
+	"github.com/kubernetes-sigs/controller-runtime/pkg/admission/certoutput"
+	"github.com/kubernetes-sigs/controller-runtime/pkg/client"
+	"github.com/kubernetes-sigs/controller-runtime/pkg/client/config"
+)
+
+// CertProvisioner provisions certificates for webhook configurations.
+type CertProvisioner struct {
+	Client client.Client
+	// Optional options for the CertInput
+	Options Options
+
+	once *sync.Once
+}
+
+// Options configures that CertInput and CertOutput approach to use.
+type Options struct {
+	CertsOutputOptions certoutput.Options
+	// CertInput generates certificate for a given common name.
+	CertInput certinput.CertInput
+}
+
+// getClientOrDie get the default client for the CertProvisioner.
+func getClientOrDie() client.Client {
```

Comment:
This already exists in the config package

Reply:
I can't find it.
There is no method returning `client.Client` in https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/client/config
The only relevant thing I can find is `client.New()` method at  https://godoc.org/github.com/kubernetes-sigs/controller-runtime/pkg/client#New